src/nvim/ex_docmd.c

```diff
@@ -5077,7 +5077,10 @@ static int get_tabpage_arg(exarg_T *eap)
       tab_number = 0;
     } else {
       tab_number = (int)eap->line2;
-      if (!unaccept_arg0 && *skipwhite((char_u *)(*eap->cmdlinep)) == '-') {
+      char *cmdp = strstr(*eap->cmdlinep, eap->cmd);
```

Comment:
You should be able to set a variable equal to `eap->cmd` and decrement it directly instead of having to use `strstr` since the cmdline string seems to stay intact and `eap->cmd` is just set to point to the beginning of the command in the cmdline string.